### PR TITLE
`Primer::ButtonComponent` to `Primer::Beta::Button` Migration Guide

### PR DESCRIPTION
**Pull request by mxriverlynn**
### Description

this PR updates the `migrations` page on the primer.style site, to include migration guides that show a developer how to move away from a deprecated component. The first guide is moving from `Primer::ButtonComponent` to `Primer::Beta::Button`. the deprecation configuration for `Primer::ButtonComponent` is also updated to include the new guide url, and mark the component as not being auto-correctable

There was also a duplicate of the Deprecated Components linter, between rubocop and erblint. this PR corrects that by disabling this linter within erblint. see the section on "Updating default erblint", below

### Screenshots

New `Migrations` page:
image

`ButtonComponent` to `Beta::Button` migration page:
image

### Updating Default ERBLint

The default ERBLint configuration runs the Deprecated Components linter twice: once as rubocop and once as erblint. this PR also updates the erblint config to not duplicate that linter.

Duplicated linter message:
image

Corrected linter message:
image

Code annotation w/ deprecation messages:
image

### Integration

> Does this change require any updates to code in production?

no

### Merge checklist

- [ ] Added/updated tests
- [x] Added/updated documentation
- [ ] Added/updated previews

### CODE REVIEW

**Review by langermank — COMMENTED** [review]
This is looking great! There's a few things we might want to add:

- `scheme` `:outline` was removed, and we recommend using `:invisible`
- I think maybe the `trailing_visual` returned a counter component, but now you can pass in an icon or a counter. Do we need to document how to use the new slots, or maybe just link to Lookbook?
- For the dropdown param, we would recommend people use this markup/slot https://primer.style/view-components/lookbook/inspect/primer/beta/button/trailing_action so maybe again, we can just link to Lookbook examples 🤔

File: docs/content/guides/primer_button_component.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+---
+
+This guide will show you how to upgrade from the now deprecated
+[`Primer::ButtonComponent`](https://primer.style/view-components/components/button)
+to the latest [`Primer::Beta::Button`](https://primer.style/view-components/components/beta/button)
+component.
+
+## Arguments
+
+The following arguments for the component initialier have changed between the deprecated and newer versions
```
Comment (camertron) [inline docs/content/guides/primer_button_component.md:12]:
```suggestion
The following arguments for the component initializer have changed between the deprecated and newer versions
```

**Review by camertron — APPROVED** [review]
Awesome, thank you 🎉

**Review by langermank — APPROVED** [review]
This is great!

File: .erb-lint.yml [hunk 2]
```diff
@@ -39,6 +39,6 @@ linters:
       AllCops:
         DisabledByDefault: true
       Primer/DeprecatedComponents:
-        Enabled: true
+        Enabled: false
```
Comment (jonrohan) [inline .erb-lint.yml:42]: Curious about this change

Comment (mxriverlynn) [reply .erb-lint.yml:42]: this is already run in rubocop, so it's just duplicated here and didn't seem necessary. the PR description shows the duplicate output when this was running vs non-duplicate output after i disabled it

**Pull request merged**